src/lib.rs

```diff
@@ -678,8 +678,16 @@ impl Builder {
     ///
     /// The file `name` will be added to the clang arguments.
     pub fn header_contents(mut self, name: &str, contents: &str) -> Builder {
+        // Apparently clang relies on having virtual FS correspondent to
+        // the real one, so we need absolute paths here
```

Comment:
Is this really needed? It seems a bit finicky. Can we file an upstream issue at least?

Reply:
Yeah, I agree this may seem kinda weird, but unfortunately, after spending quite a bit of time debugging surprisingly complicated clang's argument parsing system I have not obtained a full understanding about the true nature of this requirement :(
So probably this change could be merged with some "HACK" or "ToDo" marker while I check this up on clang mailing lists or somewhere else?
And regarding adding a test - yep, sure, I'll do that.

Thanks!

Reply:
Btw, I've asked clang guys on the mailing list [](http://lists.llvm.org/pipermail/cfe-dev/2020-May/065488.html), but have not received any response yet :(

Reply:
@emilio Hey, I've taken a look at adding the test, and unfortunately, I'm not sure what is the right way to do that in this case: as I understand, tests use bindgen as a standalone app and pass additional parameters to it via "// bindgen-flags:" comment, but I don't see a way to pass "header_contents" parameter like that, so I'd appreciate a bit of advice. Thanks!

Reply:
We have tests for these APIs in `tests/tests.rs`. They're not auto-generated but it seems relatively straight-forward.

Sorry for the lag getting back to you on this.

Reply:
Well, this is really sad, but apparently, I've broken the regular "header_contents" option. I have not figured out yet what's going wrong now, but I hope to find time to get back to it in the next few days.

Reply:
Btw, @emilio, are there any plans to show errors returned from clang in a more explicit way? :) Because currently debugging such an issue means building debug version of libclang and debugging libclang itself, which is not too convenient. I've seen that on the upper layers of clang information returned on errors of this kind is really scarce, but probably there are any ideas how to work it around anyway? Thanks!

Reply:
Errors from clang can be obtained using `TranslationUnit::diags`, not sure that's hwat you meant though.